src/main/java/wolox/training/models/Book.java

```diff
@@ -81,6 +84,7 @@ public String getGenre() {
     }
 
     public void setGenre(String genre) {
+        Preconditions.checkNotNull(genre);
```

Comment:
This one is nullable. I would only check that , IF it's not null, then check that it's not empty.